File: Backend/models/Pricing.js
const mongoose = require('mongoose');

const PricingSchema = new mongoose.Schema({
  storeId: { type: String, index: true },
  sku: { type: String, index: true },
  productName: String,
  price: Number,
  date: { type: Date, index: true }
}, { timestamps: true });

module.exports = mongoose.model('Pricing', PricingSchema);

File: Frontend/lib/features/pricing/domain/entities/pricing.dart
class Pricing {
  final String id;
  final String storeId;
  final String sku;
  final String productName;
  final double price;
  final DateTime date;
  final bool isSynced;

  Pricing({
    required this.id,
    required this.storeId,
    required this.sku,
    required this.productName,
    required this.price,
    required this.date,
    this.isSynced = true,
  });
}